k2/python/csrc/torch/v2/autograd/utils.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * @copyright
+ * Copyright      2021  Xiaomi Corp.  (authors: Wei Kang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef K2_PYTHON_CSRC_TORCH_V2_AUTOGRAD_UTILS_H_
+#define K2_PYTHON_CSRC_TORCH_V2_AUTOGRAD_UTILS_H_
+
+#include <utility>
+
+#include "k2/python/csrc/torch/v2/ops.h"
+#include "k2/python/csrc/torch/v2/ragged_arc.h"
+
+namespace k2 {
+
+class PhantomSetScoresFunction
+    : public torch::autograd::Function<PhantomSetScoresFunction> {
+ public:
+  static torch::Tensor forward(torch::autograd::AutogradContext *ctx,
+                               RaggedArc &fsa,
```

Comment:
This function uses only `fsa.Scores()`.

Can we pass it directly?

Reply:
And also for the following functions.

Reply:
No, I tried it. When I  passed a tensor, the autograd did not work.